locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -924,9 +924,10 @@ protected MachineLocation obtainOnce(ConfigBag setup) throws NoMachinesAvailable
                         executeCommandThrowingOnError(
                                 (SshMachineLocation)machineLocation,
                                 "Generate hostname " + node.getName(),
-                                Arrays.asList("sudo hostname " + node.getName(),
-                                        "sudo sed -i \"s/HOSTNAME=.*/HOSTNAME=" + node.getName() + "/g\" /etc/sysconfig/network",
-                                        "sudo bash -c \"echo 127.0.0.1   `hostname` >> /etc/hosts\"")
+                                ImmutableList.of(BashCommands.chainGroup(
+                                        "sudo bash -c \"echo 127.0.0.1   " + node.getName() + " >> /etc/hosts\"",
```

Comment:
prefer `"echo 127.0.0.1 " + node.getName() + " | sudo tee -a /etc/hosts"` to this